src/features/game/expansion/components/resources/oilReserve/OilReserve.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useState } from "react";
+import { RecoveredOilReserve } from "./components/RecoveredOilReserve";
+import { Context } from "features/game/GameProvider";
+import { MachineState } from "features/game/lib/gameMachine";
+import { OilReserve as IOilReserve } from "features/game/types/game";
+import { useSelector } from "@xstate/react";
+import Decimal from "decimal.js-light";
+import { getTimeLeft } from "lib/utils/time";
+import {
+  OIL_RESERVE_RECOVERY_TIME,
+  canDrillOilReserve,
+} from "features/game/events/landExpansion/drillOilReserve";
+import useUiRefresher from "lib/utils/hooks/useUiRefresher";
+import { RecoveringOilReserve } from "./components/RecoveringOilReserve";
+import { DepletedOilReserve } from "./components/DepletedOilReserve";
+
+interface Props {
+  id: string;
+}
+
+const _reserve = (id: string) => (state: MachineState) =>
+  state.context.state.oilReserves[id];
+const _drills = (state: MachineState) =>
+  state.context.state.inventory["Oil Drill"] ?? new Decimal(0);
+
+const compareResource = (prev: IOilReserve, next: IOilReserve) => {
+  return JSON.stringify(prev) === JSON.stringify(next);
+};
+
+export const OilReserve: React.FC<Props> = ({ id }) => {
+  const { gameService } = useContext(Context);
+  const [drilling, setDrilling] = useState(false);
+  const [oiledAmount, setOilAmount] = useState<number>();
+
+  const reserve = useSelector(gameService, _reserve(id), compareResource);
+  const drills = useSelector(gameService, _drills);
+  const timeLeft = getTimeLeft(
```

Comment:
On this function the second argument takes Seconds, and the const being passed is in Milliseconds. 
Giving my a recovery time of 833 days.